Upgrade from 1.x to Cloud (#2424)
influxdata/docs-v2: content/influxdb/cloud/upgrade/v1-to-cloud/_index.md

7. [Migrate continuous queries](#migrate-continuous-queries)
8. [Collaborate with other users](#collaborate-with-other-users)

## Create an InfluxDB Cloud account
Do one of the following to create an InfluxDB Cloud account:

- [Subscribe through InfluxData](/influxdb/cloud/get-started/#subscribe-through-influxdata) and
  [start for free](/influxdb/cloud/get-started/#start-for-free).
- [Subscribe through your cloud provider](/influxdb/cloud/get-started/#subscribe-through-a-cloud-provider).

## Create an All-Access authentication token
InfluxDB Cloud requires all requests to be authenticated with **token authentication**.
Create an **All-Access** token in your InfluxDB Cloud user interface (UI) to use
for the upgrade process.

1. Click **Data (Load Data) > Tokens** in the left navigation bar.

    {{< nav-icon "data" >}}
2. Click **{{< icon "plus" >}} Generate**, and then select **All-Access Token**.
3. Enter a description for the token, and then click **{{< icon "check" >}} Save**.

_For more information about managing tokens and token types, see [Manage tokens](/influxdb/cloud/security/tokens/)._

## Download and install the influx CLI

> Visit the [InfluxDB downloads page](https://portal.influxdata.com/downloads/)

Reviewer: ```suggestion
1. Visit the [InfluxDB downloads page](https://portal.influxdata.com/downloads/)
```

and download the **InfluxDB Cloud CLI** (`influx`).

> Place the `influx` binary in your system `PATH` or execute the CLI commands from

Reviewer: ```suggestion
2. Place the `influx` binary in your system `PATH` or execute the CLI commands from
```

Reviewer: Would it be helpful to provide a command to put the binary in your PATH here? I feel like we've done this somewhere else?

Author: This process and the `PATH` location is OS-specific and I didn't want to introduce more OS complexity here.

the directory where the `influx` CLI exists.

> [Create a CLI connection configuration](/influxdb/cloud/reference/cli/influx/#provide-required-authentication-credentials)

Reviewer: ```suggestion
3. [Create a CLI connection configuration](/influxdb/cloud/reference/cli/influx/#provide-required-authentication-credentials)
```

for your InfluxDB Cloud account.
Include the following flags:

- **-\-config-name**:
  Unique name for the connection configuration.
- **-\-host-url**:
  [InfluxDB Cloud region URL](/influxdb/cloud/reference/regions/).
- **-\-org**:
  InfluxDB Cloud organization name.
  The default organization name is the email address associated with your account.
- **-\-token**:
  InfluxDB Cloud **All-Access** token.

```sh
influx config create \
  --config-name cloud \
  --host-url https://cloud2.influxdata.com \
  --org your.email@example.com \
  --token mY5uP3rS3cRe7Cl0uDt0K3n \
  --active

> ```

Reviewer: indent for numbered list

{{% note %}}
#### Required InfluxDB Cloud credentials
All `influx` CLI examples below assume the required InfluxDB Cloud **host**,
**organization**, and **authentication token** credentials are provided by your
[`influx` CLI configuration](/influxdb/cloud/reference/cli/influx/#provide-required-authentication-credentials).

> {{% /note %}}

Reviewer: Good best practice! Perhaps we could link back to this note in context of examples? If folks land on "Create DBRP mappings," for example, from elsewhere and haven't set up their config, they may not see this note...think we've had a couple users run into this issue. I know this is a global issue and hairy one to address.

Author: I added a new caption shortcode that lets you put a caption under images and code blocks. I used this under each of the v2 CLI examples.

## Create DBRP mappings
InfluxDB database and retention policy (DBRP) mappings associate database and
retention policy combinations with InfluxDB cloud [buckets](/influxdb/cloud/reference/glossary/#bucket).
These mappings allow InfluxDB 1.x clients to query and write to
InfluxDB Cloud buckets while using the 1.x DBRP convention.

_For more information about DBRP mapping, see
[Database and retention policy mapping](/influxdb/cloud/reference/api/influxdb-1x/dbrp/)._

> **To map a DBRP combination to an InfluxDB Cloud bucket:**

Reviewer: ```suggestion
**To map a DBRP combination to an InfluxDB Cloud bucket**
```

    - _(optional)_ `--default` flag if you want the retention policy to be the default retention
      policy for the specified database

    {{< code-tabs-wrapper >}}
    {{% code-tabs %}}
[DB with one RP](#)
[DB with multiple RPs](#)
    {{% /code-tabs %}}
    {{% code-tab-content %}}
```sh
influx v1 dbrp create \
  --db example-db \
  --rp example-rp \
  --bucket-id 00xX00o0X001 \
  --default
```
    {{% /code-tab-content %}}
    {{% code-tab-content %}}
```sh
# Create telegraf/autogen DBRP mapping with autogen
# as the default RP for the telegraf DB

influx v1 dbrp create \
  --db telegraf \
  --rp autogen \
  --bucket-id 00xX00o0X001 \
  --default

# Create telegraf/downsampled-daily DBRP mapping that
# writes to a different bucket

influx v1 dbrp create \
  --db telegraf \
  --rp downsampled-daily \
  --bucket-id 00xX00o0X002
```
    {{% /code-tab-content %}}
    {{< /code-tabs-wrapper >}}

## Dual write to InfluxDB 1.x and InfluxDB Cloud
Update external clients to write to your InfluxDB Cloud instance.
**We recommend writing data to both InfluxDB 1.x and InfluxDB Cloud until you
finish [migrating your existing time series data](#migrate-time-series-data)**.

> Configure external clients with your InfluxDB Cloud **host**, **organization**,

Reviewer: ```suggestion
1. Configure external clients with your InfluxDB Cloud **host**, **organization**,
```

Author: I didn't combine this item with the process below because this refers to more than just Telegraf. This could be client libraries, other write queues like Kafka, etc. Telegraf is just the most common, which is why I included the Telegraf update process specifically.

and **authentication token**.

> ### Update Telegraf configurations
> If using Telegraf to collect and write metrics to InfluxDB 1.x, update your
> Telegraf configuration to write to both InfluxDB 1.x and InfluxDB Cloud:
> 
> 1.  Update your Telegraf configuration with a `influxdb_v2` output to write to
>     your InfluxDB Cloud instance.

Reviewer: ```suggestion

2. If using Telegraf to collect and write metrics to InfluxDB 1.x, update your
Telegraf configuration to write to both InfluxDB 1.x and InfluxDB Cloud by including a `influxdb_v2` output to write to your InfluxDB Cloud instance.
```

Author: Related to the comment above.

    ##### Example dual-write Telegraf configuration
    ```toml
    # Write metrics to InfluxDB 1.x
    [[outputs.influxdb]]
      urls = ["https://localhost:8086"]
      database = "example-db"
      retention_policy = "example-rp"

    # Write metrics to InfluxDB Cloud
    [[outputs.influxdb_v2]]
      urls = ["https://cloud2.influxdata.com"]
      token = "$INFLUX_TOKEN"
      organization = "your.email@example.com"
      bucket = "example-db/example-rp"
    ```

> 2.  Add the `INFLUX_TOKEN` environment variable to your Telegraf environment(s)
>     and set the value to your InfluxDB Cloud authentication token.
> 
> 3.  Restart Telegraf with the updated configuration and begin writing to both
>     InfluxDB 1.x and InfluxDB Cloud.
> 

Reviewer: ```suggestion
3.  Add the `INFLUX_TOKEN` environment variable to your Telegraf environment(s)
    and set the value to your InfluxDB Cloud authentication token.

4.  Restart Telegraf with the updated configuration and begin writing to both
    InfluxDB 1.x and InfluxDB Cloud.

```

Author: Related to the comment above.

1. Use the **InfluxDB 1.x** [`influx_inspect export` command](/{{< latest "influxdb" "v1" >}}/tools/influx_inspect/#export)
   to export time series data as line protocol.
   Include the `-lponly` flag to exclude comments and the data definition
   language (DDL) from the output file.

   _We recommend exporting each DBRP combination separately to easily write data
   to a corresponding InfluxDB Cloud bucket._

    ```sh
    # Syntax
    influx_inspect export \
      -database <database-name> \
      -retention <retention-policy-name> \
      -out <output-file-path> \
      -lponly

    # Example
    influx_inspect export \
      -database example-db \
      -retention example-rp \
      -out /path/to/example-db_example-rp.lp \
      -lponly
    ```

2. Use the **InfluxDB Cloud** [`influx write` command](/influxdb/cloud/reference/cli/influx/write/)
   to write the exported line protocol to InfluxDB Cloud.

    ```sh
    # Syntax
    influx write \
      --bucket <bucket-name> \
      --file <path-to-line-protocol-file>

    # Example
    influx write \
      --bucket example-db/example-rp \
      --file /path/to/example-db_example-rp.lp
    ```

> Repeat this process for each bucket.

Reviewer: ```suggestion
3. Repeat steps 1-2 for each bucket.
```
